documentation/cluster-controller/cluster-controller.adoc

```diff
@@ -0,0 +1,267 @@
+can be changed through a corresponding environment variable.
+
+When a new ConfigMap is created on the Kubernetes/OpenShift cluster, the controller gets the deployment configuration from
+its `data` section and starts the creation process of a new cluster (Kafka or Kafka Connect). It's done creating a bunch 
+of Kubernetes/OpenShift resources needed for such a deployment, like StatefulSets, ConfigMaps, Services and so on.
+
+Every time the ConfigMap is updated by the user with some changes in the `data` section, the controller executes a related updating
+operation on the cluster (i.e. scale up/down, changing metrics configuration and so on). It means that some resources are 
+patched or deleted and then re-created for reflecting the changes in the cluster configuration. 
+
+Finally, if the ConfigMap is deleted, the controller starts to un-deploy the cluster deleting all the related Kubernets/OpenShift
+resources.
+
+== Reconciliation
+
+When the controller is up and running, it reacts at all notifications received from the Kubernetes/OpenShift cluster related 
+to the watched ConfigMaps when they are created, updated or deleted.
+
+It could happen that the cluster controller isn't working properly and the related pod is restarted. In that scenario,
+it could lost notifications about the cluster ConfigMaps; without any action there will be an inconsistent state comparing
+the ConfigMaps and what is actually deployed in the Kubernetes/OpenShift cluster.
+
+For fixing this kind of issues, a periodic reconciliation is executed by the cluster controller so that it can compare 
+how the ConfigMaps look like (and if they exist) with the current cluster deployment in order to have a consistent state 
+across all of them.
+
+== Format of the cluster ConfigMap
+
+By default, the controller watches for ConfigMaps having the label `strimzi.io/kind=cluster` in order to find and get
+configuration for a Kafka or Kafka Connect cluster to deploy. The label is configurable through the `STRIMZI_CONFIGMAP_LABELS` 
+environment variable.
+
+In order to distinguish which "type" of cluster to deploy, Kafka or Kafka Connect, the controller checks the
+`strimzi.io/type` label which can have the following values :
+
+* `kafka`: the ConfigMap provides configuration for a Kafka cluster (with Zookeeper ensemble) deployment
+* `kafka-connect`: the ConfigMap provides configuration for a Kafka Connect cluster deployment
+* `kafka-connect-s2i`: the ConfigMap provides configuration for a Kafka Connect cluster deployment using Build and Source2Image
+features (works only with OpenShift)
+
+The `data` section of such ConfigMaps contains different keys depending on the "type" of deployment as described in the 
+following paragraphs.
+
+=== Kafka
+
+In order to configure a Kafka cluster deployment, it's possible to specify the following fields in the `data` section of 
+the related ConfigMap :
+
+* `kafka-nodes`: number of Kafka broker nodes
```

Comment:
Default values should be listed for all fields where applicable.

Reply:
The number of Zookeeper nodes is different between ephemeral (1) and persistent (3) so there is no a well defined default value. Do we want to update the ephemeral to have 3 Zookeeper nodes ?

Reply:
That is not really true. The default is only one and it is independent on the storage type. We have 2 example config maps with two different settings ... but that is not the default which will be used when you don't specify `kafka-nodes` at all. These should list the defaults from the code, not from the example config maps or OpenShift templates.